api/src/main/java/io/minio/MinioClient.java

```diff
@@ -913,32 +916,56 @@ private HttpResponse execute(Method method, String region, String bucketName, St
       request = Signer.signV4(request, region, accessKey, secretKey);
     }
 
-    if (this.traceStream != null) {
-      this.traceStream.println("---------START-HTTP---------");
-      String encodedPath = request.httpUrl().encodedPath();
-      String encodedQuery = request.httpUrl().encodedQuery();
-      if (encodedQuery != null) {
-        encodedPath += "?" + encodedQuery;
+    Response response = null;
+
+    // Loop to retry with new location of 307.
+    for (int i = 0; i < TEMP_REDIRECT_TRY_COUNT; i++) {
```

Comment:
I agree, I assumed request could be made outside the for loop but that is not right. This is fine.